codeserver-python/rockcraft.yaml

```
@@ -0,0 +1,246 @@
+
+      # Run the miniforge installer
+      /bin/sh miniforge.sh -b -f -p "${CONDA_DIR}"
+
+
+      # Configure conda
+      "${CONDA_BIN}/conda" config --system --set auto_update_conda false
+      "${CONDA_BIN}/conda" config --system --set show_channel_urls true
+
+      echo "conda ${MINIFORGE_VERSION:0:-2}" >> "${CONDA_DIR}/conda-meta/pinned"
+      echo "python ==${PYTHON_VERSION}" >> "${CONDA_DIR}/conda-meta/pinned"
+
+      # Install the correct versions of python, conda, pip
+      "${CONDA_BIN}/conda" install -y -q \
+                          python="${PYTHON_VERSION}" \
+                          conda="${MINIFORGE_VERSION:0:-2}" \
+                          pip="${PIP_VERSION}"
+
+      # Clean up
+      "${CONDA_BIN}/conda" update -y -q --all
+      "${CONDA_BIN}/conda" clean -a -f -y
+
+      # install - ipykernel
+      # NOTE: we need this for jupyter codeserver extension to work
+      echo "ipykernel ==${IPYKERNEL_VERSION}" >> ${CONDA_DIR}/conda-meta/pinned
+      "${CONDA_BIN}/conda" install -y -q ipykernel==${IPYKERNEL_VERSION}
+
+      # Clean up
+      "${CONDA_BIN}/conda" clean -a -f -y
+
+
+      # install - requirements.txt
+      cp ${CRAFT_PART_SRC}/components/example-notebook-servers/codeserver-python/requirements.txt /tmp/requirements.txt
+      "${CONDA_BIN}/pip" install --no-cache-dir -r /tmp/requirements.txt
+      rm -f /tmp/requirements.txt
+
+
+      # install - codeserver extensions
+      code-server --install-extension "ms-python.python@${CODESERVER_PYTHON_VERSION}" --force 
+      code-server --install-extension "ms-toolsai.jupyter@${CODESERVER_JUPYTER_VERSION}" --force 
+      code-server --list-extensions --show-versions
+
+
+      # Create some directories for staging files
+      mkdir -p "${CRAFT_PART_INSTALL}/${CONDA_DIR}" \
+      "${CRAFT_PART_INSTALL}/etc" \
+      "${CRAFT_PART_INSTALL}/${HOME}" \
+      "${CRAFT_PART_INSTALL}/${HOME_TMP}" \
+      ${CRAFT_PART_INSTALL}/lib \
+      "${CRAFT_PART_INSTALL}/usr/bin" \
+      ${CRAFT_PART_INSTALL}/${XDG_RUNTIME_DIR}
+
+      # Write some config files for activating conda in user sessions
+      chmod 2775 ${CRAFT_PART_INSTALL}/${CONDA_DIR}
+      echo ". /opt/conda/etc/profile.d/conda.sh" >> "${HOME}/.bashrc"
+      echo ". /opt/conda/etc/profile.d/conda.sh" >> "${CRAFT_PART_INSTALL}/etc/profile"
+      echo "conda activate base" >> "${HOME}/.bashrc"
+            echo "conda activate base" >> "${CRAFT_PART_INSTALL}/etc/profile"
+
+      # Do some cleanup
+      rm -rf "${HOME}/.wget-hsts" "${HOME}/.cache"
+
+      cp -r ${CRAFT_PART_SRC}/components/example-notebook-servers/codeserver-python/home ${CRAFT_PART_INSTALL}
+
+      # Stage directories
+      cp -r /lib/code-server-${CODESERVER_VERSION} ${CRAFT_PART_INSTALL}/lib/code-server
+      ln -sf /lib/code-server/bin/code-server ${CRAFT_PART_INSTALL}/usr/bin/code-server
+      cp -r "${CONDA_DIR}" "${CRAFT_PART_INSTALL}/opt"      
+      cp -r "${HOME}" "${CRAFT_PART_INSTALL}/home"
```

issue: We want the contents of $HOME to go under the `jovyan` user:
```suggestion
      cp -r "${HOME}" "${CRAFT_PART_INSTALL}/home/${NB_USER}"
```

---

It seams to be working in the original way:
```
(base) root@f321d063cf44:/# ls -la /home/jovyan/
total 40
drwxrwsr-x 1 _daemon_ root 4096 Apr 18 12:49 .
drwxr-xr-x 1 root     root 4096 Apr 18 12:43 ..
-rw-r--r-- 1 _daemon_ root   56 Apr 18 12:37 .bashrc
drwxr-sr-x 3 _daemon_ root 4096 Apr 18 12:49 .cache
drwxr-xr-x 2 _daemon_ root 4096 Apr 18 12:43 .conda
drwxr-xr-x 3 _daemon_ root 4096 Apr 18 12:43 .config
drwxr-xr-x 1 _daemon_ root 4096 Apr 18 12:43 .local
(base) root@f321d063cf44:/# ls -la /tmp_home/jovyan/
total 24
drwxrwsr-x 5 _daemon_ root 4096 Apr 18 12:44 .
drwxr-xr-x 3 root     root 4096 Apr 18 12:43 ..
-rw-r--r-- 1 _daemon_ root   56 Apr 18 12:37 .bashrc
drwxr-xr-x 2 _daemon_ root 4096 Apr 18 12:44 .conda
drwxr-xr-x 3 _daemon_ root 4096 Apr 18 12:43 .config
drwxr-xr-x 3 _daemon_ root 4096 Apr 18 12:43 .local
(base) root@f321d063cf44:/# 
```